src/DynamoDb.DistributedLock/DynamoDbDistributedLock.cs

```diff
@@ -17,18 +19,22 @@ public class DynamoDbDistributedLock : IDynamoDbDistributedLock
     private readonly IAmazonDynamoDB _client;
     private readonly DynamoDbLockOptions _options;
     private readonly Lazy<IRetryPolicy> _retryPolicy;
+    private readonly ILockMetrics _lockMetrics;
 
     /// <summary>
     /// Initializes a new instance of the <see cref="DynamoDbDistributedLock"/> class.
     /// </summary>
     /// <param name="client">The DynamoDB client.</param>
     /// <param name="options">Configuration options for the lock.</param>
+    /// <param name="lockMetrics">Collects telemetry based on lock operations</param>
     public DynamoDbDistributedLock(IAmazonDynamoDB client,
-        IOptions<DynamoDbLockOptions> options)
+        IOptions<DynamoDbLockOptions> options,
+        ILockMetrics lockMetrics)
```

Comment:
Since we register `ILockMetrics` in the service extensions, I made this non-optional to catch accidental failures to pass it in. Its a breaking change so I am not sure how you want to handle this as far as potentially making it optional and falling back to just using the static `LockMetrics.Default` I added.